### DISCUSSION TITLE
Having multiple loggers as default_logger?

### DISCUSSION THREAD

**Discussion by bheemann** [gabime/spdlog#2444] [Q&A] [answered]
Hello people,

I am using default_logger API (spdlog::info(...), etc.) in my whole project.

The default_logger implements an async_logger with console and file sink.

Now I want to add a TCP sink. However, the associated TCP server is not always reachable, so my other sinks are currently blocking for the timeout period for the TCP sink.

To prevent this, I have spun off the TCP sink into its own async_logger, so that the sink is processed in its own thread.

This meets my requirements so far.

But I don't want to change the places where the default_logger API is used in the project and I want to have two loggers that are fed by the default_api.

Is that possible? Am I following a completely wrong approach? I would be happy about any suggestions!

Greetings

**Comment by gabime** [comment, marked as answer]
A possible solution is to have a custom sink that forwards the logs to 2 async loggers, each with its own dedicated thread pool  (async tcp+async file/stdout).

So the final setup would be 3 loggers:
* Default logger with custom sink (sync logger, no need for async here).
* Couple of async loggers


The downside is some performance hit  since the data needs to be duplicated to 2 queues.

**Reply by bheemann** [reply]
I have not implemented my own custom sink yet. Therefore, such a solution approach has not entered my head. It sounds absolutely like the right way.

Thanks for the quick reply.

**Comment by nitingaikwad-28** [comment]
I am facing below issue with spdlogger.
In one process , I create spd logger with some file name using Sink vector. But when I try to create spd logger second time in same process with different file name, not able to see the logs into file when below line is called second time with different - logger name.
std::make_shared<spdlog::logger>(spdlog::logger("loggername2", sink.begin(),sink.end())

Thanks in advance for help !!

**Reply by tt4g** [reply]
#2864
